### ISSUE TITLE
Trigger unity script compilation

### ISSUE THREAD

**Issue by Merichbier** [microsoft/vscode-dotnettools#591]
Type: <b>Feature Request</b>

When working with assembly definitions, and adding a new file under an assembly, it is required to switch to the unity window so that the script compilation is triggered. Only then is the new class analyzed for errors / references.
It would be great if the script compilation is triggered on file save.

Extension version: 0.9.1
VS Code version: Code 1.83.0 (e7e037083ff4455cf320e344325dacb480062c3c, 2023-10-03T16:12:16.321Z)
OS version: Windows_NT x64 10.0.22621
Modes:

EDIT : this is for the Unity plugin.. I don't know why the issue reporter did not mention it.

**Comment by sailro** [comment]
Hello @Merichbier,

Could you try to enable `Refresh Unity's Asset Database when saving a C# script`, in `Settings/Extensions/Unity`

Thanks!

**Comment by jbevain** [comment]
Note: we should also refresh the assetdatabase when a `.asmdef` is modified/renamed/created/deleted/etc.

**Comment by Merichbier** [comment]
Hi @sailro,

Thanks for the hint, I don't know how I missed this setting. However, I've enabled it, it still doesn't seem to trigger any assembly compilation, so I do not have code analysis and IntelliSense proposing interfaces / classes in the same assembly.

**Comment by jbevain** [comment]
Hi @Merichbier,

For this feature to work there's a few pre-requisites:

* Unity needs to be open on your project
* The `Visual Studio Editor` package should be installed in your Unity project (and above 2.0.20 for the Visual Studio Code integration to work properly, as detailed in the README of the extension).
* Visual Studio Code and Unity need to be able to communicate through UDP.

With this setting enabled, VSCode will send through UDP a request for Unity to regenerate the project files. VSCode should pick the new changes that will include your newly created files and you should get completion.

If that doesn't work, we'd love to investigate why.

As a workaround, you can also add new files through the Solution Explorer in Visual Studio Code.

**Comment by jbevain** [comment]
I think we're in a position to close this:

* We're now refreshing the asset database when adding/moving/changing asmdef in 0.9.2. 
* We have an option to refresh on saving a .cs or .asmdef file.

If you're still encountering issue, please provide more information, thank you!

**Comment by Merichbier** [comment]
@jbevain Thanks for the updates. Sorry for the late answer, I was quite busy. I had the chance to test it once more and at first it didn't work. I then looked a bit into my settings, and I had "Omnisharp: Use Modern Net" setting disabled. Once I enabled it, it works as you described. I'm a bit surprised as it is stated that it does not support non-SDK-style .NET Framework projects, including Unity.
If it is expected to enable this setting, it would be good to mention it in the Unity extension details.
Thanks a lot for the time you took into investigating the issue and improvement with the .asmdef file! This is really appreciated!
